std-bits/base/src/main/java/org/enso/base/ObjectComparator.java

```diff
@@ -0,0 +1,125 @@
+package org.enso.base;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.Comparator;
+import java.util.function.BiFunction;
+
+public class ObjectComparator implements Comparator<Object> {
+  private static ObjectComparator INSTANCE;
+
+  public static ObjectComparator getInstance(BiFunction<Object, Object, Integer> fallbackComparator) {
+    if(INSTANCE == null) {
+      INSTANCE = new ObjectComparator(fallbackComparator);
+    }
+
+    return INSTANCE;
+  }
```

Comment:
This seems dangerous because on the second call the `fallbackCompataror` will be ignored. What if a different one is provided than the first time?

Would at least suggest a comment saying that the same one is always provided.